src/main/java/org/tugraz/sysds/runtime/data/DenseBlock.java

```diff
@@ -59,7 +59,23 @@ protected DenseBlock(int[] dims) {
 		//materialize dim offsets (reverse cumprod)
 		_odims = createDimOffsets(dims);
 	}
-	
+
+	/**
+	 * Get the ith dimensions size of the dense block.
+	 *
+	 * @param i the number of dimension to get
+	 * @return the size of the dimension
+	 */
+	public final int getDim(int i) {
+	    if (i == 0) {
+	    	return _rlen;
+		}
+	    if (i == _odims.length) {
+	    	return _odims[i - 1];
+		}
+	    return _odims[i - 1] / _odims[i];
```

Comment:
1) The formatting tabs vs spaces is off
2) Maybe rewrite to single return and nested ternary operators (to improve readability)